### ISSUE TITLE
[Bug] Non evicted cache

### ISSUE THREAD

**Issue by LNow** [hirosystems/clarinet#70]
I made some changes in contract, ran `clarinet test` and some tests failed. Rolled back all changes and clarinet still sees code prior rollback. Deleted whole directory and cloned project repository once again - clarinet still sees old code.
Cleared `/tmp`, performed full system restart - clarinet still sees old code.
Cloned same repo to directory with different name - everything works exactly like it should

```sh
lnow@W540:~/Projects/citycoin$ RUST_BACKTRACE=1 clarinet test

Running file:///home/lnow/Projects/citycoin/tests/core.test.ts
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: ("Analysis error: use of unresolved contract 'ST1HTBVD3JG9C05J7HBJTHGR0GGW7KXW28M5JS8QE.citycoin-core'", Some(Diagnostic { level: Error, message: "use of unresolved contract 'ST1HTBVD3JG9C05J7HBJTHGR0GGW7KXW28M5JS8QE.citycoin-core'", spans: [Span { start_line: 1, start_column: 2, end_line: 1, end_column: 15 }, Span { start_line: 1, start_column: 17, end_line: 1, end_column: 71 }, Span { start_line: 1, start_column: 74, end_line: 1, end_column: 88 }, Span { start_line: 1, start_column: 90, end_line: 1, end_column: 130 }], suggestion: None }))', src/test/deno.rs:854:22
stack backtrace:
   0: rust_begin_unwind
             at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/std/src/panicking.rs:493:5
   1: core::panicking::panic_fmt
             at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/core/src/panicking.rs:92:14
   2: core::result::unwrap_failed
             at /rustc/53cb7b09b00cbea8754ffb78e7e3cb521cb8af4b/library/core/src/result.rs:1355:5
   3: clarinet::test::deno::sessions::perform_block
   4: deno_core::ops_json::op_sync::{{closure}}
   5: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
   6: deno_core::ops::OpTable::route_op
   7: deno_core::bindings::opcall
   8: <extern "C" fn(A0) .> R as rusty_v8::support::CFnFrom<F>>::mapping::c_fn
   9: _ZN2v88internal25FunctionCallbackArguments4CallENS0_15CallHandlerInfoE
             at ./../../../../v8/src/api/api-arguments-inl.h:158:3
  10: _ZN2v88internal12_GLOBAL__N_119HandleApiCallHelperILb0EEENS0_11MaybeHandleINS0_6ObjectEEEPNS0_7IsolateENS0_6HandleINS0_10HeapObjectEEESA_NS8_INS0_20FunctionTemplateInfoEEENS8_IS4_EENS0_16BuiltinArgumentsE
             at ./../../../../v8/src/builtins/builtins-api.cc:113:36
  11: _ZN2v88internalL26Builtin_Impl_HandleApiCallENS0_16BuiltinArgumentsEPNS0_7IsolateE
             at ./../../../../v8/src/builtins/builtins-api.cc:143:5
  12: Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
fatal runtime error: failed to initiate panic, error 5
Aborted
lnow@W540:~/Projects/citycoin$ grep -r "citycoin-core" *
lnow@W540:~/Projects/citycoin$
```

**Comment by whoabuddy** [comment]
I can confirm I had a similar issue - created a branch where I was updating some error codes in the contracts, and all tests were updated and passing.

After switching back to the `main` branch without the changes, running `clarinet test` showed it was comparing my new error codes that didn't exist yet to what was in the contract, causing it to fail.

I also tried cloning the repo into a directory with a new name and had no issues on the `main` branch, all tests performed as expected.
